packages/protocol/contracts/governance/Governance.sol

```diff
@@ -649,7 +672,71 @@ contract Governance is IGovernance, Ownable, Initializable, UsingLockedGold, Ree
   }
 
   /**
-   * @notice Withdraws refunded Celo Gold commitments.
+   * @notice Whitelists the hash of a hotfix.
+   * @param txHash The abi encoded keccak256 hash of the hotfix transaction to be whitelisted.
+   * @param validatorIndex The index in the validator set of the whitelisting validator.
+   */
+  function whitelistHotfix(bytes32 txHash, uint256 validatorIndex) external nonReentrant {
+    if (msg.sender == approver) {
+      require(!hotfixes[txHash].approved);
+      hotfixes[txHash].approved = true;
+    } else {
+      IValidators validators = IValidators(registry.getAddressForOrDie(VALIDATORS_REGISTRY_ID));
+      require(msg.sender == validators.validatorAddressFromCurrentSet(validatorIndex));
```

Comment:
Since we're doing this check at `isHotfixPassing`, do we need to restrict at all who can call `whitelistHotfix`?

Reply:
great point

Reply:
making this non-restrictive causes the event log to include any accounts (not just previous/current validators/approver)